src/metabase/api/search.clj

```diff
@@ -81,19 +81,21 @@
    :updated_at          :timestamp
    ;; returned for Card only
    :dashboardcard_count :integer
-   :dataset_query       :text
    :moderated_status    :text
    ;; returned for Metric and Segment
    :table_id            :integer
-   :database_id         :integer
    :table_schema        :text
    :table_name          :text
    :table_description   :text
+   ;; returned for Metric, Segment, and Action
+   :database_id         :integer
    ;; returned for Database and Table
    :initial_sync_status :text
    ;; returned for Action
    :model_id            :integer
-   :model_name          :text))
+   :model_name          :text
+   ;; returned for Card and Action
+   :dataset_query       :text))
```

Comment:
it's a little comical to annotate with all the "returned for Model X and Model Y', but I guess for now it's worth keeping it in. If I'm still bothered by this next time I see it I'll pull it out into an ASCII table in a comment.